viper.go

```diff
@@ -1976,7 +1976,7 @@ func (v *Viper) searchInPath(in string) (filename string) {
 		}
 	}
 
-	if b, _ := exists(v.fs, filepath.Join(in, v.configName)); b {
+	if b, _ := exists(v.fs, filepath.Join(in, v.configName)); b && v.configType != "" {
```

Comment:
It's a minor point for performance-wise I'd do the single string check first. Otherwise LGTM.